instrumentation/aws_sdk/lib/opentelemetry/instrumentation/aws_sdk/handler.rb

```diff
@@ -9,51 +9,34 @@ module Instrumentation
     module AwsSdk
       # Generates Spans for all interactions with AwsSdk
       class Handler < Seahorse::Client::Handler
-        def call(context)
-          span_name = get_span_name(context)
-          attributes = get_span_attributes(context)
+        def call(context) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+          return super unless context
 
-          tracer.in_span(span_name, kind: OpenTelemetry::Trace::SpanKind::CLIENT, attributes: attributes) do |span|
-            execute = proc {
-              super(context).tap do |response|
-                if (err = response.error)
-                  span.record_exception(err)
-                  span.status = Trace::Status.error(err)
-                end
-              end
-            }
+          service_name = context.client.class.api.metadata['serviceId'] || context.client.class.to_s.split('::')[1]
+          operation = context.operation&.name
+          attributes = {
+            'aws.region' => context.config.region,
+            OpenTelemetry::SemanticConventions::Trace::RPC_SYSTEM => 'aws-api',
+            OpenTelemetry::SemanticConventions::Trace::RPC_METHOD => operation,
+            OpenTelemetry::SemanticConventions::Trace::RPC_SERVICE => service_name
+          }
+          attributes[SemanticConventions::Trace::DB_SYSTEM] = 'dynamodb' if service_name == 'DynamoDB'
```

Comment:
I removed the db helper as this one liner didn't seem like it needed a module.  I think we can abstract this portion away if/when a need arises.   It also prevents from creating an additional hash to be merged.

Reply:
ideally this instrumentation has many modules with many helpers. IIRC there is sns/sqs additions coming shortly, and we should assume that we'll need a pattern to handle enriching the span attributes with extra information depending on the service, ala node https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-aws-sdk/src/services . That being said, fine to kick the can on that.

Reply:
I separated it to different module in order to be able to extend the basic client implementation and to add attributes for specific AWS services.
I'm working on another PR to support message attributes for SQS/SNS, including context propagation. 

I'm fine with this implementation for now, soon I'll open another PR with the ability to extend aws services.